pkg/api/controllers/runner/provider/list.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package provider
+
+import (
+	"fmt"
+	"net/http"
+
+	_ "github.com/daytonaio/daytona/pkg/models"
+	"github.com/daytonaio/daytona/pkg/server"
+	"github.com/gin-gonic/gin"
+)
+
+// ListProviders godoc
+//
+//	@Tags			provider
+//	@Summary		List providers
+//	@Description	List providers
+//	@Produce		json
+//	@Success		200	{array}	models.ProviderInfo
+//	@Router			/runner/provider [get]
+//
+//	@id				ListProviders
+func ListProviders(ctx *gin.Context) {
```

Comment:
Maybe we can have a `runnerId` optional query param instead of 2 distinct methods